vacasa/connect/connect.py

```diff
@@ -481,3 +481,129 @@ def get_offices(self, params: dict = None):
+            departure: Checkout date in 'YYYY-MM-DD' format
+            children: How many children will be staying
+            pets: How many pets will be staying
+            trip_protection: Has the user requested trip protection?
+                -1 No
+                 0 TBD
+                 1 Yes
+
+        Returns: dict
+
+        """
+        url = f"{self.endpoint}/v1/quotes"
+        headers = self._headers()
+
+        params = {
+            'adults': adults,
+            'children': children,
+            'pets': pets,
+            'unit_id': unit_id,
+            'arrival': arrival,
+            'departure': departure,
+        }
+
+        if trip_protection is not None:
+            params[trip_protection] = trip_protection
+
+        return self._get(url, headers, params).json()
+
+    def create_reservation(self,
+                           unit_id: int,
+                           arrival: str,
+                           departure: str,
+                           email: str,
+                           address: dict,
+                           adults: int,
+                           quote_id: str,
+                           first_name: str,
+                           last_name: str,
+                           account_number: str,
+                           exp_mmyy: str,
+                           phone: Optional[str] = None,
+                           children: int = 0,
+                           pets: int = 0,
+                           trip_protection: Optional[bool] = None,
+                           source: Optional[str] = None
+                           ) -> dict:
+        """ Reserve a given unit
+
+        Arguments:
+            unit_id: A Vacasa Unit ID
+            arrival: Checkin date in 'YYYY-MM-DD' format
+            departure: Checkout date in 'YYYY-MM-DD' format
+            email: User's email address
+            phone: User's phone number
+            address: User's address information, e.g.
+                {
+                    'address_1': '999 W Main St #301',
+                    'city': 'Boise',
+                    'state': 'ID',
+                    'zip': '83702'
+                }
+            adults: How many adults will be staying
+            children: How many children will be staying
+            pets: How many pets will be staying
+            trip_protection: Has the user requested trip protection?
+                -1 No
+                 0 TBD
+                 1 Yes
+            quote_id: ID of a quote retrieved from the `GET /quotes` endpoint
+            first_name: User's First Name (for billing)
+            last_name: User's Last Name (for billing)
+            account_number: Credit card #
+            exp_mmyy: Credit card expiry in `mmyy` format
+            source: A Vacasa-issued code identifying the source of this request
+
+        Returns: dict
+
+        """
+
+        url = f"{self.endpoint}/v1/reservations"
```

Comment:
I'm wondering if the SDK should validate that the endpoint scheme is https in the `__init__` now that we're going to be passing through credit card numbers. What do you think?

Reply:
good call!  added a commit to address this